### PR TITLE
updates for Processing 4

### PR DESCRIPTION
**Pull request by benfry**
Changes included: 

* Changes to use reflection to access the deprecated/removed `frame` field
* Some whitespace was trimmed automatically by my editor; feel free to ignore
* Should work fine with both Processing 4.x and 3.x

### CODE REVIEW

**Comment by gaocegege** [comment]
Thanks for the PR! @jeremydouglass Please have a look.

**Review by gaocegege — APPROVED** [review]

**Comment by jeremydouglass** [comment]
Sorry for the delay -- been extremely ill, catching up.

**Comment by jeremydouglass** [comment]
Looks like I also still need to migrate this repo from Travis-CI to GitHub actions. sigh.

File: src/test/e2e/util/ImageUtils.java [hunk 1]
```diff
@@ -29,12 +29,12 @@ private static float imgDifference(PImage i0, PImage i1) {
     i1.loadPixels();
     int[] ip1 = i1.pixels;
     for (int n = 0; n < ip0.length; n++) {
-      int pxl0 = ip0[n]
```
Comment (jeremydouglass) [inline src/test/e2e/util/ImageUtils.java:32]: That's weird. Blame says that I introduced this breaking typo when I merged a large set of codacy style changes -- 

https://github.com/processing-r/Processing.R/blame/733fb86e5a695eb78d26082cb189098f3b901b21/src/test/e2e/util/ImageUtils.java#L32

but then that build passed on the CI server!

https://travis-ci.org/github/processing-r/Processing.R/builds/569579118

I'm not sure how that is possible, unless that build didn't perform the e2e ImageUtils phase at all. Something to look at when setting up Actions.

**Review by jeremydouglass — COMMENTED** [review]
I think I can see that Processing 4 has moved to PConstants.MACOS, e.g.
https://github.com/processing/processing4/commit/fab6b59c1b06c430a1f78885d7f3ca7667b5c977

Going forward, any reason we should add *both* MACOSX / MACOS internally to our available PConstants? Currently we just import PConstants from the processing 3 core, attach them to BuiltinApplet and then extend RLangPApplet with them.

File: src/test/e2e/util/ImageUtils.java (hunk 1 above)
Comment (jeremydouglass) [reply src/test/e2e/util/ImageUtils.java:32]: Nevermind, that build actually failed in the log, it just displayed as passed for some reason. ugh.

**Comment by benfry** [comment]
Nope, `MACOSX` to `MACOS` is a change in name only. So you can keep it `MACOSX` for now if you want to use the same source with 3.x and 4.x (that's why I reverted it back to `MACOSX` here).

That being said, I wouldn't recommend supporting 3.x any longer. Apple's aggressive OS changes alone will make it nearly impossible to run Processing there in the near future, if not already.

**Comment by gaocegege** [comment]
Any progress here? I can help test it if needed.

**Comment by jeremydouglass** [comment]
@gaocegege sorry this dropped off my radar. I've build and run through the examples / tests -- everything still works on Processing 3, but could you give the release I upload a quick try on the latest Processing 4? I don't have a 10.14 mac available at the moment to confirm.

**Comment by jeremydouglass** [comment]
Closed #262

**Pull request merged**